compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -4279,6 +4282,13 @@ private STNode parseTerminalExpression(SyntaxKind kind, STNode annots, boolean i
                 return parseTransactionalExpression();
             case SERVICE_KEYWORD:
                 return parseServiceConstructorExpression(annots);
+            case BASE16_KEYWORD:
+            case BASE64_KEYWORD:
+                nextNextToken = getNextNextToken(kind);
+                if (nextNextToken.kind == SyntaxKind.BACKTICK_TOKEN) {
```

Comment:
I think we can skip this check, since `base16` and `base64` keywords only occurs in expressions. 
Then if one of those two, then we can commit to say its one of those two literals.